### PR TITLE
RP-461 Tenant Administration - DataSource password handling API

### PR DESCRIPTION
**Pull request by vgonzalez9**
completed password unmasking changes and encrypting changes,
Password unmasking only applies to the tenant specific dataSources
Added new default logic to return  nulls for Default Properties :
      DataSources : database, username, password and schemaSearchPath
Added new getResources class to read in json files for json results - and show the urls used to create the file.
updated existing .json test data with latest tenant property updates
Updated test cases.
Added enums for rdw services and for their DataSource names
Removed Mock data from Tenant. Still in sandbox atm.
Cleaned up log messages
Fix /configprops test file to use the correct properties for tenant-configuration-lookup:

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/client/DefaultConfigServerClient.java [hunk 1]
```diff
@@ -34,7 +35,7 @@ public void postMonitor() {
             headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
             HttpEntity<String> request = new HttpEntity<>("path=*", headers);
             ResponseEntity<String> responseEntity = restTemplate.postForEntity(configServerUri + "/monitor", request, String.class);
-            logger.debug("Successfully called {}/monitor returned {}.", configServerUri, responseEntity.getStatusCode());
+            logger.info("Successfully called {}/monitor returned {}.", configServerUri, responseEntity.getStatusCode());
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/client/DefaultConfigServerClient.java]: probably back to debug for this message

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/client/DefaultConfigServerClient.java [hunk 2]
```diff
@@ -43,21 +44,41 @@ public void postMonitor() {
...
-            String errorMsg = String.format("Cannot contact %s/%s/awsuat at this time.", configServerUri, service);
+            String errorMsg = String.format("Cannot contact %s/%s/* at this time.", configServerUri, service);
             logger.error(errorMsg, e);
             throw new IllegalArgumentException(errorMsg);
         }
     }
 
+    @Override
+    public String getActualReportingROPasswordProps() throws IllegalArgumentException {
+        return getActualServicePassword("rdw-reporting-report-processor");
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/client/DefaultConfigServerClient.java]: ENUM or constant perhaps?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultTenantConfigPropsService.java [hunk 3]
```diff
@@ -40,17 +44,30 @@
...
         List<String> configProps = tenantConfigurationLookupClient.getRawConfigurations();
+        // getting props with actual unmasked passwords
+        final String reportingROProps = defaultConfigServerClient.getActualReportingROPasswordProps();
+        configProps.add(reportingROProps);
+//        final String reportingRWProps = defaultConfigServerClient.getActualReportingRWPasswordProps();
+//        configProps.add(reportingRWProps);
+//        final String olapProps = defaultConfigServerClient.getActualOlapROPasswordProps();
+//        configProps.add(olapProps);
+//        final String warehouseProps = defaultConfigServerClient.getActualWarehouseRWPasswordProps();
+//        configProps.add(warehouseProps);
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultTenantConfigPropsService.java]: clean up old code

File: admin-service/src/test/java/org/opentestsystem/rdw/admin/client/DefaultConfigServerClientTest.java [hunk 4]
```diff
@@ -57,18 +57,26 @@ public void itShouldEncryptPasswordAndSuccessfullyDecryptIt() {
         assertThat(decryptedPassword).isNotNull().isEqualTo(password);
     }
 
-    @Ignore
+    //@Ignore
```
Comment (ftdholbrook) [inline admin-service/src/test/java/org/opentestsystem/rdw/admin/client/DefaultConfigServerClientTest.java]: remove the ignore instead of commenting out

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/client/ConfigServerClient.java [hunk 5]
```diff
@@ -14,11 +14,35 @@
 
     /**
      * @param service    to search through for the dataSource's password
-     * @param dataSource of the password - need to be the property
      * @return the actual encrypted password of the dataSource for the service
      * @throws IllegalArgumentException not able to to call service endpoint
      */
-    String getActualServicePassword(String service, String dataSource) throws IllegalArgumentException;
+    String getActualServicePassword(String service) throws IllegalArgumentException;
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/client/ConfigServerClient.java]: i like the javadocs for the exceptions but it isn't necessary to declare unchecked exceptions in the method signature

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/client/RDWServices.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package org.opentestsystem.rdw.admin.client;
+
+/**
+ * Used to share the RDW Service Names
+ */
+public enum RDWServices {
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/client/RDWServices.java]: @vgonzalez9 thanks for adding this

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/client/RDWServices.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    RDW_REPORTING_SERVICE("rdw-reporting-service"),
+    RDW_REPORTING_REPORT_AGGREGATE("rdw-reporting-report-aggregate"),
```
Comment (malaffoon) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/client/RDWServices.java]: why is this not "rdw-reporting-aggregate-service"?

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/service/RdwServiceConfigPropsAccessor.java [hunk 8]
```diff
@@ -75,4 +75,28 @@
      * @return a map of each tenant and its AggregateReportingProperties
      */
     Map<String, AggregateReportingPropertiesTenant> getTenantsAggregateReportingProperties();
+    /**
+     * @param key of the tenant to lookup for the reporting_ro datasource
+     * @return the unmasked, unencrypted password for the report_ro DataSource
+     */
+    Optional<String> getActualReportingROPassword(final String key);
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/service/RdwServiceConfigPropsAccessor.java]: nit: interface doesn't need final

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/RDWDataSourceNames.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    REPORTING_RO("reporting_ro"),
+    REPORTING_RW("reporting_rw"),
+    WAREHOUSE_RW("warehouse_rw"),
+    OLAP_RO("olap_ro");
+
+    private String name;
+    RDWDataSourceNames(final String name) {
+        this.name = name;
+    }
+    public String getName(){
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/service/impl/RDWDataSourceNames.java]: i would be tempted to call this an "ID"/ `getId()` just to not make this confusing with the native enum `name()` method

File: admin-service/src/test/java/org/opentestsystem/rdw/admin/service/impl/TestConfigPropResources.java [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+        return rootNode.toString();
+    }
+
+    /**
+     * @return the results actuatorReportingAggregateProperties.json file.
+     *         Contains results from a call to http://localhost:8308/configprops
+     * @throws Exception From getFile or mapper
+     */
+    private static String getActuatorReportingAggregateServiceProperties() throws Exception {
+        File file = ResourceUtils.getFile("classpath:actuatorReportingAggregateProperties.json");
```
Comment (esotrope) [inline admin-service/src/test/java/org/opentestsystem/rdw/admin/service/impl/TestConfigPropResources.java]: 👍

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/client/RDWServices.java [hunk 11]
```diff
@@ -0,0 +1,17 @@
+package org.opentestsystem.rdw.admin.client;
+
+/**
+ * Used to share the RDW Service Names
+ */
+public enum RDWServices {
+    RDW_REPORTING_SERVICE("rdw-reporting-service"),
+    RDW_REPORTING_REPORT_AGGREGATE("rdw-reporting-aggregate-service"),
+    RDW_REPORTING_REPORT_PROCESSOR("rdw-reporting-report-processor");
+    private String name;
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/client/RDWServices.java]: same comment - would call this `id`

File: admin-service/src/main/resources/application.yml [hunk 12]
```diff
@@ -197,14 +197,7 @@ reporting:
...
-tenant-configuration-lookup:
-  services:
-    admin-service:
-      url: http://localhost:8208
-    aggregate-service:
-      url: http://localhost:8308
-    report-processor:
-      url: http://localhost:8086
-    reporting-service:
-      url: http://localhost:8108
```
Comment (malaffoon) [inline admin-service/src/main/resources/application.yml]: i can see removing the aggregate-service but doesn't deleting these mean a developer has to configure things to work locally? A general philosophy we have is that, all other things being equal, it's nice to make it so a developer can pull stuff down and run things. An example is the application.yml for the webapp. Is it that this admin stuff is just so ornery that it isn't worth it?

File: admin-service/src/test/resources/application-tenant_client_docker.yml [hunk 13]
```diff
@@ -2,13 +2,9 @@ logging:
...
-      url: http://localhost:8308
-    report-processor:
-      url: http://localhost:8086
-    reporting-service:
-      url: http://localhost:8108
\ No newline at end of file
+    admin-service: http://localhost:8208
+    aggregate-service: http://localhost:8308
+    report-processor: http://localhost:8086
+    reporting-service: http://localhost:8108
```
Comment (malaffoon) [inline admin-service/src/test/resources/application-tenant_client_docker.yml]: Why this change? I guess i don't really care if there is a good reason but it makes it inconsistent with other yml entries (and i have to say our yml entries have gotten a little out of hand).
